gym/spaces/dict.py

```diff
@@ -1,10 +1,12 @@
 from collections import OrderedDict
 from collections.abc import Mapping, Sequence
+from typing import Optional, Union
+import typing as t
```

Comment:
Not a huge fan of this import, I'm assuming it's for `t.Dict`? I think it'd be better to do something like `from typing import Dict as DictType` or something like that; or just `typing.Dict` in the code.

Or commit to the newer style and use `dict` for type hints, see my later comments

Reply:
I didn't realize `dict` works with __future__ annotations in 3.7. Will use that.